utils/dependency_management.py

```diff
@@ -294,10 +294,20 @@ def check_requirements(extras, github_annotate):
 
         for requirement_abstract in requirements_abstract:
             for requirement_concrete in requirements_concrete:
-                version = Specifier(str(requirement_concrete.specifier)).version
-                if canonicalize_name(requirement_abstract.name) == canonicalize_name(
-                    requirement_concrete.name
-                ) and requirement_abstract.specifier.contains(version):
+                if '@' in str(requirement_concrete):
+                    # `@` is not listed as a valid `specifier` in `class Specifier`.
+                    version = str(requirement_concrete).split('@')[1]
```

Comment:
Shouldn't this be the whole string? right now you only use the repo name as version, but the branch and other tags are also relevant
```suggestion
                    version = "@".join(str(requirement_concrete).split('@')[1:])
```